### PR TITLE
V1.4.1 Unit test for platform_services-handler

### PR DESCRIPTION
**Pull request by supritat**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change
Included unit testing for platform_services-handler

### CODE REVIEW

File: platform-services-handler/config/test-config.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (deepusundar) [inline platform-services-handler/config/test-config.json]: The configs are not in sync with internal version, Validate what features are missing.  For example:
  "SERVICE_DELETION_EVENT_START": "CALL_DELETE_WORKFLOW",
  "SERVICE_DELETION_EVENT_END" : "DELETE_PROJECT",

File: platform-services-handler/index.js [hunk 2]
```diff
@@ -117,25 +117,26 @@ module.exports.handler = (event, context, cb) => {
```
Comment (deepusundar) [inline platform-services-handler/index.js]: Make sure all service handler functionalities  from Internal version is ported

File: platform-services-handler/test/test.js [hunk 3]
```diff
@@ -15,12 +15,344 @@
...
+      var callFunction = index.handler(event, context, callback);
+      var logCheck = logStub.args[0][0].includes(logMessage);
+      reqStub.restore();
+    };
+    AWS.restore("SQS");
+    logStub.restore();
+    assert.isTrue(logCheck);
+  });
+
+  it("should indicate sucess from 200 response for the kinesis data with completed endingEvent",()=>{
```
Comment (deepusundar) [inline platform-services-handler/test/test.js]: Kinesis input validation cases are covered. We need to also add test cases specific to the handler functionalities. Like how different Events are handled, what is the back-end service being triggered, How failure events are queued etc?

Comment (deepusundar) [reply platform-services-handler/test/test.js]: You may need to refactor the index.js file to add unit test cases to cover all scenarios.

File: platform-services-handler/components/config.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 // =========================================================================
-// Copyright � 2017 T-Mobile USA, Inc.
+// Copyright � 2017 T-Mobile USA, Inc.
```
[the remark this answers is not in the document]
Comment (supritat) [reply platform-services-handler/components/config.js]: Corrected the copyright symbol

File: platform-services-handler/components/config.js [hunk 5]
```diff
@@ -33,6 +33,8 @@ var getStageConfig = (context) => {
     configObj = require('../config/stg-config.json');
   } else if (functionName.endsWith('prod')) {
     configObj = require('../config/prod-config.json');
+  } else if (functionName.endsWith('test')){
```
[the remark this answers is not in the document]
Comment (supritat) [reply platform-services-handler/components/config.js]: config file made consistent with other services

File: platform-services-handler/index.js [hunk 6]
```diff
@@ -316,6 +311,7 @@ module.exports.handler = (event, context, cb) => {
 												"failure_message": null
 											});
 											logger.info("created a new service in service catalog.");
+											logger.verbose("created a new service in service catalog.");
```
[the remark this answers is not in the document]
Comment (supritat) [reply platform-services-handler/index.js]: log info with same details are removed

File: platform-services-handler/index.js [hunk 7]
```diff
@@ -714,6 +710,7 @@ module.exports.handler = (event, context, cb) => {
 						}
 
 					} else {
+						logger.error('push un-interesting event to processed queue');
```
[the remark this answers is not in the document]
Comment (supritat) [reply platform-services-handler/index.js]: replaced error log with debug log

File: platform-services-handler/index.js [hunk 8]
```diff
@@ -761,8 +758,10 @@ module.exports.handler = (event, context, cb) => {
 				}
 			], function (err, results) {
 			if (err) {
+				logger.info(err)
 				cb(err);
 			} else {
+				logger.verbose('events failed'+ failedEvents.length+'processed events'+processedEvents.length);
```
[the remark this answers is not in the document]
Comment (supritat) [reply platform-services-handler/index.js]: done

File: platform-services-handler/test/test.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 // =========================================================================
-// Copyright � 2017 T-Mobile USA, Inc.
+// Copyright � 2017 T-Mobile USA, Inc.
```
[the remark this answers is not in the document]
Comment (supritat) [reply platform-services-handler/test/test.js]: copyright symbol is corrected

File: platform-services-handler/test/test.js [hunk 10]
```diff
@@ -15,12 +15,344 @@
...
+    logStub = sinon.stub(logger, 'verbose', spy);
+    getReqStub = sinon.stub(request,'Request', (obj)=>{
+      return obj.callback(null, responseObjectGET, bodyObj);
+    });
+    var callFunction = index.handler(event, context, callback);
+    logStub.restore();
+    getReqStub.restore();
+    assert.isTrue(true)
+  });
+
```
[the remark this answers is not in the document]
Comment (supritat) [reply platform-services-handler/test/test.js]: updated the code with valid check condition

File: platform-services-handler/config/test-config.json (hunk 1 above)
Comment (supritat) [reply platform-services-handler/config/test-config.json]: At present Service deletion events feature is not included

File: platform-services-handler/index.js (hunk 2 above)
Comment (supritat) [reply platform-services-handler/index.js]: All the features are not merged from internal version

File: platform-services-handler/test/test.js (hunk 3 above)
Comment (supritat) [reply platform-services-handler/test/test.js]: refactoring of index.js is not included in this story

Comment (suryajak) [reply platform-services-handler/test/test.js]: We need to get to 90% code coverage, so if that requires code refactoring, we will need to do that.

**Comment by suryajak** [comment]
Please resolve the merge conflict and update the PR

**Pull request merged**